### ISSUE TITLE
pwsh ignores -OutputFormat text when writing to a redirected stderr stream

### ISSUE THREAD

**Issue by brantb** [PowerShell/PowerShell#5912]
Steps to reproduce
------------------

```powershell
$p = [Diagnostics.Process]::new()
$p.StartInfo.FileName = "pwsh"
$encoded = [Convert]::ToBase64String([Text.Encoding]::Unicode.GetBytes("throw 'boom'"))
$p.StartInfo.Arguments = "-EncodedCommand $encoded -ExecutionPolicy Bypass -NoLogo -NonInteractive -NoProfile -OutputFormat text"
$p.StartInfo.UseShellExecute = $false
$p.StartInfo.RedirectStandardError = $true
$p.Start() | Out-Null
$p.StandardError.ReadToEnd()
```

Expected behavior
-----------------
The error is written to stderr as text:
```none
boom
At line:1 char:1
+ throw 'boom'
+ ~~~~~~~~~~~~
+ CategoryInfo          : OperationStopped: (boom:String) [], RuntimeException
+ FullyQualifiedErrorId : boom
```
Changing `$p.StartInfo.RedirectStandardError` to `$false` (the default) will result in the expected output. 

Actual behavior
---------------
The error record is encoded as CLIXML before being written to stderr:
```none
#< CLIXML
<Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><S S="Error">boom_x000D__x000A_</S><S S="Error">At line:1 char:1_x000D__x000A_</S><S S="Error">+ throw 'boom'_x000D__x000A_</S><S S="Error">+ ~~~~~~~~~~~~_x000D__x000A_</S><S S="Error">+ CategoryInfo          : OperationStopped: (boom:String) [], RuntimeException_x000D__x000A_</S><S S="Error">+ FullyQualifiedErrorId : boom_x000D__x000A_</S><S S="Error"> _x000D__x000A_</S></Objs>
```

This problem doesn't seem to apply to stdout. Setting `$p.StartInfo.RedirectStandardOutput = $true` in a similar fashion works as expected.

Environment data
----------------
```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.0.0
PSEdition                      Core
GitCommitId                    v6.0.0
OS                             Microsoft Windows 10.0.16299
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by bugoutfr** [comment]
Hi,
I have the same problem here. The parent Powershell script is launched by Gitlab Runner which doesn't output to the standard output.
Regards.

**Comment by MartinSGill** [comment]
Any progress on this? It's really annoying in our build logs. I'd love to try and fix this myself.. but I've no idea where to look :(.

**Comment by MartinSGill** [comment]
Seems to have happened before and seems related to -EncodedCommand

https://github.com/hashicorp/packer/issues/4322

**Comment by SteveL-MSFT** [comment]
Looking in the [code](https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs#L1214) it appears this is by-design, but I don't know why this is the case.  @lzybkr do you recall or know why we force XML in this case?

**Comment by lzybkr** [comment]
I don't recall any history (the code comes from V1 I believe, before my time).

I'm guessing that code ignored the case where `OutputFormat` was specified **explicitly**. It makes sense to implicitly change to XML if `OutputFormat` is not specified - with the assumption being that you're piping PowerShell to PowerShell.

**Comment by SteveL-MSFT** [comment]
Ok, that makes sense.  So I can just modify it so that if `OutputFormat` was specified explicitly, we should honor that.

**Comment by TishSerg** [comment]
How about this fix for PowerShell 5.1.* branch?
